src/mumble/AudioInput.h

```diff
@@ -137,6 +137,7 @@ class AudioInputRegistrar {
 	virtual void setDeviceChoice(const QVariant &, Settings &) = 0;
 	virtual bool canEcho(const QString &outputsys) const       = 0;
 	virtual bool canExclusive() const;
+	virtual bool ifOSPermissionDenied() = 0;
```

Comment:
From the name of the function I can't entirely deduce what any given implementation is supposed to do.

Thus I suggest
1. Add some documentation
2. Maybe find a name that is more descriptive (I will help with that once I understood what the function is supposed to do :D)

Reply:
Yeah... I know I have a really bad sense for naming things. Do you have any suggestions for the name of this function?

Reply:
I might have once you explained to me what it should do ^^

Is this a query for whether the OS currently denied any permission (or specifically the permission to use the mic) or is it rather a callback that gets called whenever the before-mentioned situation is encountered?

Reply:
Both are correct.

For Mac, there's a system interface to query the permission. So this function will actually query the mic permission.

But for WASAPI, one can only see if it is denied by initializing the audio device, so I actually used a static variable to store a flag (indicates that whether this initialization failed due to permission problem) and this function will return the  value of this flag.

I know...  Nasty thing... But without refactoring the audio back end, I can't see a better way.

Reply:
> Both are correct.

From your description that doesn't seem to be true :eyes: 
In both cases this function is used to query whether the permission is granted or not but with the exception that for WASAPI this only returns a meaningful value once the audio system has been initialized.

So in that case I think we should name this function e.g. `isMicrophoneAccessPermitted()` or even `isMicrophoneAccessPermittedByOS()`

Furthermore given that special care has to be taken for WASAPI as to when to call this function, we definitely need explanatory comments that contain this kind of information.

Reply:
Sorry that I misunderstood the latter half of your last comment...

Yes, you are right. Personally, I prefer the first one `isMicrophoneAccessPermitted()` since I don't see other factors other than OS that can deny access to microphones.

I will document its implementation in WASAPI backend since the that is somehow not straight-forward.

Reply:
> Sorry that I misunderstood the latter half of your last comment...

No problem :) 

> since I don't see other factors other than OS that can deny access to microphones.

Well without having any context, it could be assumed that Mumble itself provides a way to grant/deny access to the mic. Thus making explicit that this relates to restriction from the OS probably doesn't hurt :thinking: 

> I will document its implementation in WASAPI backend since that is somehow not straight-forward.

Sounds good. But please also document at the declaration of this function that in general it must not be called before the audio system has initialized (and then refer to the docs in the WASAPI implementation :point_up: )